public/app/api/clients/folder/v1beta1/index.ts

```diff
@@ -1,8 +1,170 @@
+      newData = {
+        canAdmin: resultAccess.data.canAdmin,
+        canDelete: resultAccess.data.canDelete,
+        canEdit: resultAccess.data.canEdit,
+        canSave: resultAccess.data.canSave,
+        accessControl: resultAccess.data.accessControl,
+        created: result.data.metadata.creationTimestamp || '0001-01-01T00:00:00Z',
+        createdBy:
+          (createdBy && resultUserDisplay.data.display[resultUserDisplay.data.keys.indexOf(createdBy)]?.displayName) ||
+          'Anonymous',
+        // Does not seem like this is set to true in the legacy API
+        hasAcl: false,
+        id: parseInt(result.data.metadata.labels?.[DeprecatedInternalId] || '0', 10) || 0,
+        parentUid: result.data.metadata.annotations?.[AnnoKeyFolder],
+        managedBy: result.data.metadata.annotations?.[AnnoKeyManagerKind] as ManagerKind,
+
+        title: result.data.spec.title,
+        uid: result.data.metadata.name!,
+        updated: result.data.metadata.annotations?.[AnnoKeyUpdatedTimestamp] || '0001-01-01T00:00:00Z',
+        updatedBy:
+          (updatedBy && resultUserDisplay.data.display[resultUserDisplay.data.keys.indexOf(updatedBy)]?.displayName) ||
+          'Anonymous',
+        // Seems like this annotation is not populated
+        // url: result.data.metadata.annotations?.[AnnoKeyFolderUrl] || '',
+        // general folder does not come with url
+        // see https://github.com/grafana/grafana/blob/8a05378ef3ae5545c6f7429eae5c174d3c0edbfe/pkg/services/folder/folderimpl/folder_unifiedstorage.go#L88
+        url: uid === GENERAL_FOLDER_UID ? '' : folderUrl(result.data.metadata.name!, result.data.spec.title!),
+        version: result.data.metadata.generation || 1,
+      };
+
+      if (resultParents.data.items?.length) {
+        newData.parents = resultParents.data.items
+          .filter((i) => i.name !== result.data!.metadata.name)
+          .map((i) => ({
+            title: i.title,
+            uid: i.name,
+            // No idea how to make slug, on the server it uses a go lib: https://github.com/grafana/grafana/blob/aac66e91198004bc044754105e18bfff8fbfd383/pkg/infra/slugify/slugify.go#L56
+            // Don't think slug is needed for the URL to work though
+            url: folderUrl(i.name, i.title),
+          }));
+      }
+    }
+
+    return {
+      ...result,
+      isLoading:
+        result.isLoading ||
+        resultParents.isLoading ||
+        resultAccess.isLoading ||
+        (keys.length ? resultUserDisplay.isLoading : false),
+      isFetching:
+        result.isFetching ||
+        resultParents.isFetching ||
+        resultAccess.isFetching ||
+        (keys.length ? resultUserDisplay.isFetching : false),
+      isSuccess:
+        result.isSuccess &&
+        resultParents.isSuccess &&
+        resultAccess.isSuccess &&
+        (keys.length ? resultUserDisplay.isSuccess : true),
+      isError:
+        result.isError ||
```

Comment:
One hell of a return statement 😄  I wonder if they're all important, e.g., if fetching access info fails for some reason, does it mean that we can't use other query results?

Reply:
If the caller actually needs the access info then yes and we just don't know it here. Otherwise each part would have to be optional, which changes the return type and would need changes where this is used which I tried to avoid. It's not pretty I agree but hopefully it won't be long lived.

Reply:
I tried a bit, see in the hooks.ts file.